helpers.go

```diff
@@ -19,6 +19,12 @@ type RowData struct {
 	Values  []interface{}
 }
 
+// Idempotence. It can be either true or false, and set or not set, hence bool won't do
+type Idempotence struct {
```

Comment:
Why do we care if it is set or not? If we default to false which should be a safe value then why does it matter if a user set it explicitly to false?

Reply:
@Zariel  it's a very good question. Basically, I copied this behaviour from the Java driver which is as follows (and bool in Java can be either true/false/null):

1) If the value is either true or false, use it in the following code (in most cases in speculative queries)
2) If the value is null (that is not set), fetch the default for the cluster (which can be either true or false).

The issue we have here is that if we keep it as bool, we have no way to "initialise" it to the default value for the cluster. If a client uses Query{stmt: "something"}, idempotence will be set to false, and we have no way to say whether it's been set this way by the client or is it a default, especially as there's no action in Query value to sync with cluster's default.

Of course we can do this in the following manner: for the first release we only create the flag on the query, allow cluents to set it and default to false. A following release would add a "constructor" call for query (something like "NewQuery" ), potentially chanding/breaking the existing API, which will then allow using a cluster-wide default by querying it and setting the Query value accordingly. Then the next following release would bring "breaking" changes to Batch.Query() and Batch.Bind() APIs allowing to also use idempotence in the Batch statements. Would that make any sense? If so, I'll simplify this one and create separate tickets for the follow-ups.